linked_lists/intersection.py

```diff
@@ -1,14 +1,117 @@
 
+from os import curdir
 
 
 class Node:
     def __init__(self, value):
         self.val = value
         self.next = None
 
+# class LinkedList:
+#     def __init__(self):
+#         self.head = None
+
+#     # method to add more nodes to linked list
+#     def add_first(self, value):
+#         new_node = Node(value)
+#         new_node.next = self.head
+#         self.head = new_node
+    
+
+#     # method to return head of linked list
+#     def getHead(self):
+#         return self.head
+    
+#     def getNthNode(self, n):
+#         current = self.head
+#         while(current != None and n > 0):
+#             current = current.next
+#             n -=1;
+        
+#         return current
+
+#adding nodes to both lists
+node_d1 = Node("D")
+node_e1 = Node("E")
+node_f1 = Node("F")
+
+node_c1 = Node("C")
+node_e2 = Node("E")
+node_f2 = Node("F")
+
+#List A: [d, e1, f1]
+node_d1.next = node_e1
+node_e1.next = node_f1
+
+#List B: [c, e2, f2]
+node_c1.next = node_e2
+node_e2.next = node_f2
+node_f2.next = node_e1
+
+head_a = node_d1
+head_b = node_c1
+
+#get lenth of linked list
+def get_lin_list_len(head):
+    list_len=0
+    current=head
+
+    while current !=None:
```

Comment:
Generally, when comparing to what we call singletons such as `None` we prefer to use `is` and `not` instead of `==` and `!=`. The reason for this has to do with [operator overloading](https://www.geeksforgeeks.org/operator-overloading-in-python/) which I'm happy to answer questions about if you are curious, but it's also okay just to remember this as a best practice!

```suggestion
    while current is not None:
```